### PR TITLE
Add support for Plone 5.1.x

### PR DESCRIPTION
**Pull request by mbaechtold**

### CODE REVIEW

File: test-plone-5.1.x.cfg [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+extends =
+    https://raw.github.com/4teamwork/ftw-buildouts/master/test-plone-5.1.x.cfg
+    sources.cfg
+    versions.cfg
+
+package-name = ftw.publisher.sender
+
+
+[branches]
+ftw.publisher.core = ne/plone-5.1
```
Comment (mbaechtold) [inline test-plone-5.1.x.cfg]: 🚧

File: test-plone-5.1.x.cfg [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ftw.simplelayout = plone-5.1.x
```
Comment (mbaechtold) [inline test-plone-5.1.x.cfg]: 🚧

File: setup.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
 from setuptools import setup, find_packages
 import os
 
-version = '[1;2C.dev0'
+version = '2.7.12'
 maintainer = 'Jonas Baumann'
 
 tests_require = [
     'collective.testcaselayer',
```
Comment (jone) [inline setup.py]: I think we can remove this now, as well as `Products.PloneTestCase`.

File: ftw/publisher/sender/tests/helpers.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+        relations = [
+            RelationValue(intids.getId(aq_base(item)))
+            for item in related_items
+        ]
+
+        obj.relatedItems = relations
+
+        from zope.event import notify
+        from zope.lifecycleevent import ObjectModifiedEvent
```
Comment (jone) [inline ftw/publisher/sender/tests/helpers.py]: It's save to move these imports to the top of the file.

File: ftw/publisher/sender/tests/test_simplelayout.py [hunk 5]
```diff
@@ -3,13 +3,16 @@
...
+from ftw.publisher.sender.utils import IS_AT_LEAST_PLONE_5_1
 from ftw.simplelayout.configuration import flattened_block_uids
 from ftw.simplelayout.interfaces import IPageConfiguration
 from ftw.testbrowser import browsing
 from ftw.testing import staticuid
+from unittest2 import skipIf
 from Products.CMFCore.utils import getToolByName
 
 
+@skipIf(IS_AT_LEAST_PLONE_5_1, 'ftw.contentpage is not available for Plone 5')
```
Comment (jone) [inline ftw/publisher/sender/tests/test_simplelayout.py]: Is `TestPublishingNEWSimplelayoutTypes`, a subclass of this class, still executed under Plone 5.1? Should it be?

File: ftw/publisher/sender/tests/tests.zcml [hunk 6]
```diff
@@ -5,6 +5,7 @@
     <include package="Products.Five" file="meta.zcml" />
     <include package="Products.Five.viewlet" file="meta.zcml" />
     <include package="Products.GenericSetup" file="meta.zcml" />
+    <include package="plone.resource" file="meta.zcml" />
```
Comment (jone) [inline ftw/publisher/sender/tests/tests.zcml]: I have the suspicion that this file (`tests.zcml`) is no longer used as we have removed all PloneTestCase tests. Could you verify that?

Comment (mbaechtold) [reply ftw/publisher/sender/tests/tests.zcml]: The file is still used in https://github.com/4teamwork/ftw.publisher.sender/blob/42dae51919c9482d556f4d2f3e6df6b4a69a41de/ftw/publisher/sender/testing.py#L23 to create a `ZCML_LAYER` .

The `ZCML_LAYER` in turn is still used in https://github.com/4teamwork/ftw.publisher.sender/blob/42dae51919c9482d556f4d2f3e6df6b4a69a41de/ftw/publisher/sender/tests/test_meta_zcml.py#L14

File: ftw/publisher/sender/tests/test_simplelayout.py (hunk 5 above)
Comment (mbaechtold) [reply ftw/publisher/sender/tests/test_simplelayout.py]: I have added a commit where this is addressed.

File: setup.py (hunk 3 above)
Comment (mbaechtold) [reply setup.py]: Ok, I have removed them.

File: ftw/publisher/sender/tests/helpers.py (hunk 4 above)
Comment (mbaechtold) [reply ftw/publisher/sender/tests/helpers.py]: I have moved the imports to the top of the module.

**Pull request merged**